Workouts/SupportingViews/FilterView.swift

```diff
@@ -52,56 +68,57 @@ struct FilterView: View {
-                        Text("From")
-                    }
-                    DatePicker(selection: $endDate, in: ...(Calendar.current.date(byAdding: .month, value: 12, to: Date()) ?? Date()), displayedComponents: .date) {
-                        Text("To")
+                Section(header: ToggleableHeader(text: "Workout Distance (miles) 📏 \(workoutDistance.isApplied ? distanceString : "")", switchValue: $workoutDistance.isApplied)) {
+                    if workoutDistance.isApplied {
+                        Slider(value: $workoutDistance.value, in: 0...50)
+                            .transition(.slide)
                     }
                 }
                 
                 // Select duration
                 // Would be nice to use a countdown timer (UIDatePicker.Mode.countDownTimer)
-                // But it's not available in SwiftUI yet
-                Section(header: Text("Workout Duration ⏳")) {
-                    DatePicker(selection: $startDate, in: ...(Calendar.current.date(byAdding: .month, value: -12, to: Date()) ?? Date()), displayedComponents: .hourAndMinute) {
-                        Text("From")
-                    }
-                }
-                
-                Section(header: Text("Workout Duration ⏳")) {
-                    DatePicker(selection: $startDate, in: ...(Calendar.current.date(byAdding: .month, value: -12, to: Date()) ?? Date()), displayedComponents: .hourAndMinute) {
-                        Text("To")
+                // But it's not available in SwiftUI yet, how about a slider?
+                Section(header: ToggleableHeader(text: "Workout Duration ⏳ \(workoutDuration.isApplied ? durationString : "")", switchValue: $workoutDuration.isApplied)) {
+                    if workoutDuration.isApplied {
+                        Slider(value: $workoutDuration.value, in: 0...5)
+                            .transition(.slide)
                     }
                 }
+
                 
                 // Select calorie range
                 //
-                Section(header: Text("Calories Burned 🥵")) {
-                    Picker("From", selection: $caloriesBurnedMin) {
-                        ForEach(0 ..< 41) {
-                            Text("\($0 * 50) calories")
-                        }
-                    }
-                    Picker("To", selection: $caloriesBurnedMax) {
-                        ForEach(1 ..< 41) {
-                            Text("\($0 * 50) calories")
-                        }
+                Section(header: ToggleableHeader(text: "Calories Burned 🥵 \(caloriesBurned.isApplied ? caloriesString : "")", switchValue: $caloriesBurned.isApplied)) {
+                    if caloriesBurned.isApplied {
+                        Slider(value: $caloriesBurned.value, in: 0...2000)
+                            .transition(.asymmetric(insertion: .move(edge: .leading), removal: .slide))
```

Comment:
@emilyCheroske Would this filter workouts with calories between 0 and the slider value, or would it be more precise and only filter workouts that match the exact slider value?

Reply:
It will still have that 'range' like quality, so everything between 0 and the slider value. This makes my life a bit easier when writing the predicates that we send to health kit because I can just be like - give me back everything less than the slider value. I was just messing around with the slider widget though - I think I had thrown date pickers in there before for choosing the distance 🤣So - the user input controls are totally fluid if you want to go in there and mess around

Reply:
> > @emilyCheroske Good idea! I know you said "bare bones", but to double-check, should the filters reset when FilterView is dismissed?
> 
> I was thinking we might try and persist them actually (not in local storage or anything but just for the lifecycle of the app) - so that if you dismiss the filter page, and then re-open it, you can see the existing filters that you had already applied and tweak them if you want to modify the search results. What do you think about that?

Yeah, I think this is optimal. So far everything but the workout selection/filter persists. Additionally, the workout filter doesn't seem to work how I expect it 🤔